account_reset_chart/models/res_company.py

```diff
@@ -0,0 +1,151 @@
+                logger.info('Unlinking payment orders.')
+                self._cr.execute(
+                    """
+                    DELETE FROM payment_line
+                    WHERE order_id IN (
+                        SELECT id FROM payment_order
+                        WHERE company_id = %s);
+                    """, (self.id,))
+                self._cr.execute(
+                    "DELETE FROM payment_order WHERE company_id = %s;",
+                    (self.id,))
+
+                unlink_from_company('payment.mode')
+        except KeyError:
+            pass
+
+        unlink_from_company('account.banking.account.settings')
+        unlink_from_company('res.partner.bank')
+
+        logger.info('Unlinking reconciliations')
+        rec_obj = self.env['account.move.reconcile']
+        rec_obj.search(
+            [('line_id.company_id', '=', self.id)]).unlink()
+
+        logger.info('Reset paid invoices\'s workflows')
+        paid_invoices = self.env['account.invoice'].search(
+            [('company_id', '=', self.id), ('state', '=', 'paid')])
+        if paid_invoices:
+            self._cr.execute(
+                """
+                UPDATE wkf_instance
+                SET state = 'active'
+                WHERE res_type = 'account_invoice'
+                AND res_id IN %s""", (tuple(paid_invoices.ids),))
+            self._cr.execute(
+                """
+                UPDATE wkf_workitem
+                SET act_id = (
+                    SELECT res_id FROM ir_model_data
+                    WHERE module = 'account'
+                        AND name = 'act_open')
+                WHERE inst_id IN (
+                    SELECT id FROM wkf_instance
+                    WHERE res_type = 'account_invoice'
+                    AND res_id IN %s)
+                """, (tuple(paid_invoices.ids),))
+            paid_invoices.signal_workflow('invoice_cancel')
+
+        inv_ids = self.env['account.invoice'].search(
+            [('company_id', '=', self.id)]).ids
+        if inv_ids:
+            logger.info('Unlinking invoices')
+            self.env['account.invoice.line'].search(
+                [('invoice_id', 'in', inv_ids)]).unlink()
+            self.env['account.invoice.tax'].search(
+                [('invoice_id', 'in', inv_ids)]).unlink()
+            self._cr.execute(
+                """
+                DELETE FROM account_invoice
+                WHERE id IN %s""", (tuple(inv_ids),))
+
+        logger.info('Unlinking moves')
+        moves = self.env['account.move'].search([('company_id', '=', self.id)])
+        if moves:
+            self._cr.execute(
+                """UPDATE account_move SET state = 'draft'
+                   WHERE id IN %s""", (tuple(moves.ids),))
+        moves.unlink()
+
+        unlink_from_company('account.fiscal.position')
+        unlink_from_company('account.tax')
+        unlink_from_company('account.tax.code')
+        unlink_from_company('account.journal')
+
```

Comment:
Just an idea..redefine the unlink_from_company as:
def unlink_from_company(models):
....Call the same code but with loop for model in models....
The Call then should be : unlink_from_company([])...less lines and nice too!